### PR TITLE
(maint) Add CODEOWNERS

### PR DESCRIPTION
**Pull request by Magisus**

### CODE REVIEW

**Comment by Magisus** [comment]
Let me know if anyone else should be on here. Also if we should only be targeting master with these.

File: CODEOWNERS [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# This will cause the following groups to be assigned review
+# of any opened PRs against the branches containing this file.
+
+* @puppetlabs/puppetserver-maintainers
```
Comment (MikaelSmith) [inline CODEOWNERS]: This won't work the way you expect. See https://help.github.com/en/articles/about-code-owners#codeowners-syntax.

Comment (Magisus) [reply CODEOWNERS]: Oh I see, thanks. I assumed those were bullets.

Comment (MikaelSmith) [reply CODEOWNERS]: They're pattern matching, and last match for each file gets tagged.

**Pull request merged**